src/guiguts/tools/pptxt.py

```diff
@@ -0,0 +1,2591 @@
+        # Additionally, an emdash separating words in, for
+        # example, the ToC is replaced by a space.
+
+        # Need these twice to handle alternates
+        # E.g. fo’c’s’le
+
+        # Emdash
+        line = re.sub(
+            r"([\p{L}\p{Nd}\p{P}=])(\—)([\p{L}\p{N}’‘“_\(])",
+            mask_special_cases,
+            line,
+        )
+        line = re.sub(
+            r"([\p{L}\p{Nd}\p{P}=])(\—)([\p{L}\p{N}’‘“_\(])",
+            mask_special_cases,
+            line,
+        )
+        # Minus
+        line = re.sub(r"([\p{L}①②③④⑤\.=])(\-)([\p{L}\p{N}])", mask_special_cases, line)
+        line = re.sub(r"([\p{L}①②③④⑤\.=])(\-)([\p{L}\p{N}])", mask_special_cases, line)
+        # The others.
+        line = re.sub(r"([\p{L}①②③④⑤])(’)([\p{L}①②③④⑤])", mask_special_cases, line)
+        line = re.sub(r"([\p{L}①②③④⑤])(’)([\p{L}①②③④⑤])", mask_special_cases, line)
+        line = re.sub(r"([\p{L}①②③④⑤]|^)(’)([\p{L}①②③④⑤])", mask_special_cases, line)
+        line = re.sub(r"([\p{L}①②③④⑤]|^)(’)([\p{L}①②③④⑤])", mask_special_cases, line)
+        line = re.sub(r"([\p{L}①②③④⑤])(‘)([\p{L}①②③④⑤])", mask_special_cases, line)
+        line = re.sub(r"([\p{L}①②③④⑤])(‘)([\p{L}①②③④⑤])", mask_special_cases, line)
+        line = re.sub(r"([\p{L}①②③④⑤])(\')([\p{L}①②③④⑤])", mask_special_cases, line)
+        line = re.sub(r"([\p{L}①②③④⑤])(\')([\p{L}①②③④⑤])", mask_special_cases, line)
+        line = re.sub(r"([\p{L}①②③④⑤]|^)(\')([\p{L}①②③④⑤])", mask_special_cases, line)
+        line = re.sub(r"([\p{L}①②③④⑤]|^)(\')([\p{L}①②③④⑤])", mask_special_cases, line)
+        line = re.sub(
+            r"([\p{L}①②③④⑤])(\.)([\p{L}①②③④⑤\p{P}])", mask_special_cases, line
+        )
+        line = re.sub(
+            r"([\p{L}①②③④⑤])(\.)([\p{L}①②③④⑤\p{P}])", mask_special_cases, line
+        )
+        line = re.sub(
+            r"([\p{L}①②③④⑤])(\^)([\p{L}①②③④⑤\p{P}])", mask_special_cases, line
+        )
+        line = re.sub(
+            r"([\p{L}①②③④⑤])(\^)([\p{L}①②③④⑤\p{P}])", mask_special_cases, line
+        )
+
+        s = ""
+        indx = 0
+
+        while indx < len(line):
+            # Look for 'words', one character at a time.
+            if (
+                line[indx : indx + 1].isnumeric()
+                or line[indx : indx + 1].isalpha()
+                or line[indx : indx + 1].isspace()
+            ):
+                s = s + line[indx : indx + 1]
+            else:
+                s = s + " "
+            indx += 1
+
+        # Use temporary list to hold the split line while we put
+        # back the protected characters.
+        T = s.split()
+        L = []
+        for n in range(len(T)):
+            # For efficiency!
+            Tn = T[n]
+            Tn = Tn.replace("①", "-")
+            Tn = Tn.replace("②", "’")
+            Tn = Tn.replace("③", "‘")
+            Tn = Tn.replace("④", "'")
+            Tn = Tn.replace("⑤", ".")
+            Tn = Tn.replace("⑥", "^")
+            L.append(Tn)
```

Comment:
I may have misunderstood this little section, but would it have the same effect and possibly be more efficient and clear to omit the temporary list?
```
s.replace("①", "-")
s.replace("②", "’")
s.replace("③", "’")
s.replace("④", "’")
s.replace("⑤", "’")
s.replace("⑥", "^")
return s.split()
```

Reply:
This code extract is from the 'get_words_on_line' function. It parses the text of a line and returns a list of tokens that PPtxt considers as 'words' for the purposes of the various checks it performs. I rewrote this function immediately after PPtxt was added to the master GG. I did this to clarify and extend the range of tokens it provides, simplify the code that does it and improve its efficiency. As part of that effort the section above was coincidentally rewritten exactly in the form you suggest!

Reply:
Thanks @G4OEU - I know you have said there was additional tidying up of that nature to be done. It would be good to get a version of `pptxt` merged into `master`, so I'm happy with additional tidying going into a new PR at a later stage. I'll give @cpeel a chance to comment, but IMHO, there's nothing here that needs to stop us merging this, and allowing you to continue work in another branch.